tools/pipelines-tasks/AVDAppAttachPublish/PublishAVD.ps1

```diff
@@ -8,19 +8,49 @@
 .PARAMETER inputJsonStr
     Specifies input json configuration to trigger AppAttach Framework DLL with for AppAttach.
 
-.PARAMETER targetDLL
-    Specifies the target DLL to be triggered for AppAttach.
+.PARAMETER dllPath
+    Specifies the target DLL path used for AppAttach flow.
 #>
 
-param([string]$inputJsonStr, [string]$targetDLL)
+param([string]$inputJsonStr, [string]$dllPath)
 
-Add-Type -Path $targetDLL
+# Load the required DLLs
+Add-Type -Path (Join-Path $dllPath "AppAttachKernel.dll")
+Add-Type -Path (Join-Path $dllPath "AppAttachMessenger.dll")
```

Comment:
i feel the purpose of loading messenger dll is to get access to the interface, however i feel the interface should not be part of the implementation, please correct me if i am wrong

Reply:
As per the DLL used here, I am not able to access the `IMessageHandler` interface from solely `AppAttachKernel`. Fortunately, all DLLs are present inside the NuGet, and I am directly referencing it here to use the Message handler utilities.